src/typechecker.erl

```diff
@@ -3335,6 +3364,71 @@ pick_one_refinement_each([Ty|Tys], [RefTy|RefTys]) ->
     %% The last list is the list where to type is refined.
     RefHeadCombinations ++ RefTailCombinations.
 
+%% Is a type refinable to the point that we do exhaustiveness checking on it?
+refinable(?type(integer)) ->
+    true;
+refinable(?type(char)) ->
+    true;
+refinable(?type(non_neg_integer)) ->
+    true;
+refinable(?type(pos_integer)) ->
+    true;
+refinable(?type(neg_integer)) ->
+    true;
+refinable({atom, _, _}) ->
+    true;
+refinable(?type(nil)) ->
+    true;
+refinable(?type(union,Tys)) when is_list(Tys) ->
+    lists:all(fun refinable/1, Tys);
+refinable(?type(tuple, Tys)) when is_list(Tys) ->
+    lists:all(fun refinable/1, Tys);
+refinable({ann_type, _, [_, Ty]}) ->
+    refinable(Ty);
+refinable(_) ->
+    false.
+
+% Given a type, pick a value of that type.
+% Used in exhaustiveness checking to show an example value
+% which is not covered by the cases.
+pick_value(List) when is_list(List) ->
```

Comment:
Perhaps this function can be defined in another module? This module is huge already.

Reply:
Yeah.